### DISCUSSION TITLE
fastapi and nicegui compatibility

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [zauberzeug/nicegui#2992] [Q&A]
### Question

Hi. My app is designed to work with the latest fastapi version whenever it becomes available. I just tried to update fastapi to the latest and pip stopped with the error:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
nicegui 1.4.18 requires fastapi<0.110.0,>=0.109.1, but you have fastapi 0.110.3 which is incompatible.
```
Is nicegui compatible to specfic fastapi versions only?

**Comment by rodja** [comment]
Yes. We had some bad experience with incompatible changes of FastAPI/Starlette in the past. As I wrote in #2755 we can only upgrade to latest FastAPI version in 2.0 due to a breaking change in FastAPI 0.110.0.
